### PR TITLE
surface: add querying decode error details

### PR DESCRIPTION
**Pull request by bgrzesik**
This commit add querying decode error details using vaQuerySurfaceError.

### CODE REVIEW

File: src/surface.rs [hunk 1]
```diff
@@ -258,6 +282,61 @@ impl<D: SurfaceMemoryDescriptor> Surface<D> {
...
+        status: bindings::VAStatus,
+    ) -> Result<Vec<SurfaceDecodeMBError>, VaError> {
+        let mut raw: *const bindings::VASurfaceDecodeMBErrors = std::ptr::null();
+
+        // Safe because `self` represents a valid VASurface.
+        va_check(unsafe {
+            bindings::vaQuerySurfaceError(
+                self.display.handle(),
+                self.id,
+                status,
```
Comment (Gnurou) [inline src/surface.rs:296]: The documentation for `vaQuerySurfaceError` mentions the following:

> VA_STATUS_ERROR_DECODING_ERROR should be passed in as "error_status"

which suggests this is the only valid value for `status`, for some reason. Should we just use that and remove the `status` argument for this method?

**Review by Gnurou — APPROVED** [review]
Thanks - looking good to me, with just one small nit.

File: src/surface.rs (hunk 1 above)
Comment (Gnurou) [reply src/surface.rs:296]: Let's remove it for now then, and if new parameters arise we will add then using a well-defined enum :+1:

Comment (bgrzesik) [reply src/surface.rs]: Sure. Done 😄

**Pull request merged**